backend/cellguide/pipeline/canonical_marker_genes/canonical_markers.py

```diff
@@ -0,0 +1,285 @@
+import json
+import logging
+import pathlib
+from typing import Dict, Tuple
+
+import pandas as pd
+import requests
+
+from backend.cellguide.pipeline.canonical_marker_genes.types import (
+    AnatomicalStructure,
+    GeneBiomarker,
+    ParsedAsctbTableEntry,
+    Reference,
+)
+from backend.cellguide.pipeline.canonical_marker_genes.utils import (
+    clean_doi,
+    get_title_and_citation_from_doi,
+)
+from backend.cellguide.pipeline.constants import ASCTB_MASTER_SHEET_URL, ENSEMBL_GENE_ID_TO_DESCRIPTION_FILENAME
+from backend.wmg.data.ontology_labels import ontology_term_label
+
+logger = logging.getLogger(__name__)
+
+
+class CanonicalMarkerGenesCompiler:
+    def __init__(self, wmg_tissues: list[str], wmg_human_genes: list[str]):
+        """
+        Initializes the CanonicalMarkerGenesCompiler class.
+
+        Arguments
+        ---------
+        wmg_tissues (list[str]): List of tissue ontology term IDs present in WMG.
+        wmg_human_genes (list[str]): List of gene ensembl IDs present in WMG.
+        """
+
+        logger.info("Fetching ASCTB data...")
+        self.asctb_data = requests.get(ASCTB_MASTER_SHEET_URL).json()
+        self.wmg_tissues = [i for i in wmg_tissues if i.startswith("UBERON:")]
```

Comment:
WMG Tissues have some terms that start with "CL" because they're cell cultures. We filter these out as they are specific to our platform and don't exist in ASCTB. I will add a comment above this line.